tests/loom.rs

```diff
@@ -0,0 +1,38 @@
+#[cfg(loom)]
+#[cfg(test)]
+mod loom_tests {
+    // Evil hack to share CounterAddOp between
+    // unit tests and integration tests.
+    use left_right::Absorb;
+    include!("../src/utilities.rs");
+
+    use loom::sync::atomic::{AtomicI32, Ordering};
+    use loom::sync::Arc;
+    use loom::thread;
+
+    #[test]
+    fn read_before_publish() {
+        loom::model(|| {
+            let (mut w, r) = left_right::new::<i32, _>();
+            let answer = Arc::new(AtomicI32::new(0));
+            let answer_clone = answer.clone();
```

Comment:
You don't need this — instead, you can just return `val` from inside the spawned thread, and then get the return value out of `jh.join().unwrap()`.

Reply:
TIL! That's awesome. Updated this in a50870be1a4e7e5577084a8e109790c0373cd02e